### PR TITLE
Feature/route filters to correct service

### PR DESCRIPTION
**Pull request by mr-nick17**
### What

Added middleware to route to the existing filter service or the new filter/flex service dependent on whether the dataset type is flexible for the route `/filters`

### How to review

Sense check
Tests pass

### Who can review

Anyone but me

### CODE REVIEW

File: middleware/datasetType/handler.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+			filterID, err := helpers.ReturnSecondSegmentFromPath(path)
+			if err != nil {
+				log.Error(ctx, "failed to extract filter id info from path", err, log.Data{"filter_id": filterID, "path": path})
+				return
+			}
+
+			// Obtain access_token from cookie
+			userAccessToken := ""
+			c, err := req.Cookie(`access_token`)
```
Comment (joegasewicz) [inline middleware/datasetType/handler.go:42]: Do these multiple client calls need to be concurrent? - just a question...

**Review by joegasewicz — APPROVED** [review]
Apart from the question about multiple concurrent client calls LGTM

File: middleware/datasetType/handler.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+			if err == nil && len(c.Value) > 0 {
+				userAccessToken = c.Value
+				log.Info(req.Context(), "obtained access_token Cookie")
+			}
+
+			f, _, err := filterClient.GetJobState(ctx, userAccessToken, "", "", "", filterID)
```
Comment (joegasewicz) [inline middleware/datasetType/handler.go:48]: Do these multiple client calls need to be concurrent? - just a question...

Comment (mr-nick17) [reply middleware/datasetType/handler.go:48]: Possibly, but most are dependent on one another. You need the `filterID` to get the job state which you need to get the dataset to get the type to work out whether the type is flexible

**Pull request merged**